ts/commitments.ts

```diff
@@ -264,11 +265,51 @@ export class Batch {
     }
 }
 
+export function batchFactory(
+    batchType: Usage,
+    txDescription: TransactionDescription,
+    accountRoot: string
+): Batch {
+    if (batchType == Usage.Transfer) {
```

Comment:
We are relying on the batchType here to determine which batch class to use. 

Technically txDescription gives us the function name (like 'submitTransfer') and the function signature (such as  'submitTransfer(bytes32[],uint256[2][],uint256[],bytes[])'). We can 
1. hardcode the function names as constant. But need to remember to update when the contract changes.
2. wrangling with typechain interfaces to get the correct function name even if the contract changes. But typechain is hard to wrangle.

The batchType looks more straightforward here.